llvm/lib/DebugInfo/DWARF/DWARFUnit.cpp

```diff
@@ -658,6 +711,9 @@ void DWARFUnit::clearDIEs(bool KeepCUDie) {
   // It depends on the implementation whether the request is fulfilled.
   // Create a new vector with a small capacity and assign it to the DieArray to
   // have previous contents freed.
+  llvm::sys::ScopedWriter FreeLock(CUDieFreeMutex);
+  llvm::sys::ScopedWriter CULock(CUDieArrayMutex);
```

Comment:
I *think* taking all 3 locks is overkill; If you ahve CUDieFreeMutex, the other two are guaranteed to be available.